packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -112,19 +113,41 @@ class CallContext {
     this.callId = callId;
   }
 
+  public onCallEnded(handler: (callEndedData: CallAdapterCallEndedEvent) => void): void {
+    this.emitter.on('callEnded', handler);
+  }
+
+  public offCallEnded(handler: (callEndedData: CallAdapterCallEndedEvent) => void): void {
+    this.emitter.off('callEnded', handler);
+  }
+
   public updateClientState(clientState: CallClientState): void {
     const call = this.callId ? clientState.calls[this.callId] : undefined;
     const latestEndedCall = findLatestEndedCall(clientState.callsEnded);
-    this.setState({
-      ...this.state,
-      userId: clientState.userId,
-      displayName: clientState.callAgent?.displayName,
-      call,
-      page: getCallCompositePage(call, latestEndedCall),
-      endedCall: latestEndedCall,
-      devices: clientState.deviceManager,
-      latestErrors: clientState.latestErrors
-    });
+
+    // As the state is transitioning to a new state, trigger appropriate callback events.
+    const oldPage = this.state.page;
+    const newPage = getCallCompositePage(call, latestEndedCall);
+    if (!IsCallEndedPage(oldPage) && IsCallEndedPage(newPage)) {
```

Comment:
Let's say customer was subscribed to the `callEnded` event.
They have custom logic for showing an end call screen when this event is triggered, but they have a timeout that renders that page after 5 seconds.
They would see our end call page for 5 seconds before seeing their own right?


Just trying to make sure I understand how this fix is working...

Reply:
Yep that's correct